pkg/sql/database_region_change_finalizer.go

```diff
@@ -296,40 +297,12 @@ func (r *databaseRegionChangeFinalizer) repartitionRegionalByRowTables(
 			tabledesc.UpdateIndexPartitioning(index.IndexDesc(), index.Primary(), newImplicitCols, newPartitioning)
 		}
 
-		// Remove zone configurations that applied to partitions that were removed
-		// in the previous step. This requires all indexes to have been
-		// repartitioned such that there is no partitioning on the removed enum
-		// value. This is because `deleteRemovedPartitionZoneConfigs` generates
-		// subzone spans for the entire table (all indexes) downstream for each
-		// index. Spans can only be generated if partitioning values are present on
-		// the type descriptor (removed enum values obviously aren't), so we must
-		// remove the partition from all indexes before trying to delete zone
-		// configurations.
-		for _, index := range tableDesc.NonDropIndexes() {
-			// Remove zone configurations that reference partition values we removed
-			// in the previous step.
-			update, err := prepareRemovedPartitionZoneConfigs(
-				ctx,
-				txn,
-				tableDesc,
-				index.GetID(),
-				oldPartitionings[index.GetID()],
-				index.GetPartitioning(),
-				r.localPlanner.ExecCfg(),
-			)
-			if err != nil {
-				return nil, nil, err
-			}
-			if update != nil {
-				zoneConfigUpdates = append(zoneConfigUpdates, update)
-			}
-		}
```

Comment:
Can you explain why this can be removed? I don't understand.

Reply:
Ah... this is a bug which works by accident, more details here: https://cockroachlabs.slack.com/archives/C4ERHM60Z/p1668628589384259
The tl;dr is that the zone config updates generated from this deleted block are partial and broken, but they're overwritten by a good zone config a few lines below...